File: SMI-SDK/.NET SDK/Library/gis/trace/appender/SMI/BrokerSMILoggerAppenderConfigProperties.cs
namespace pt.sapo.gis.trace.appender.SMI
{
    using System;
    using System.Collections.Generic;
    using pt.sapo.gis.trace.configuration;

    public class BrokerSMILoggerAppenderConfigProperties : SMILoggerAppenderConfigProperties
    {
        public BrokerSMILoggerAppenderConfigProperties(IDictionary<String,String> properties) : base(properties) { }

        public String Host { get { return this["host"]; } }

        public int Port { get { return Int32.Parse(this["port"]); } }

        public String Topic { get { return this["topic"]; } }
    }
}

File: SMI-SDK/Java SDK/src/pt/sapo/gis/trace/appender/SMI/BrokerSMILoggerAppenderConfigProperties.java
package pt.sapo.gis.trace.appender.SMI;

import java.util.List;
import pt.sapo.gis.trace.configuration.Appender.Property;
import pt.sapo.gis.trace.configuration.AppenderConfigProperties;

public class BrokerSMILoggerAppenderConfigProperties extends SMILoggerAppenderConfigProperties{    
    
    
    public BrokerSMILoggerAppenderConfigProperties(List<Property> properties){         
        super(properties);
    }
    
    public String getHost(){
        return propertiesMap.get("host");
    }
            
    public Integer getPort(){        
        Integer i = null;
        if(propertiesMap.containsKey("port"))
            i = Integer.parseInt(propertiesMap.get("port"));
        return i;
    }
    
    public String getTopic(){
        return propertiesMap.get("topic");
    }
       
}
